gdx/src/com/badlogic/gdx/graphics/g3d/utils/RenderableShapeBuilder.java

```diff
@@ -0,0 +1,163 @@
+
+package com.badlogic.gdx.graphics.g3d.utils;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Mesh;
+import com.badlogic.gdx.graphics.VertexAttributes.Usage;
+import com.badlogic.gdx.graphics.g3d.Renderable;
+import com.badlogic.gdx.graphics.g3d.RenderableProvider;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.FlushablePool;
+
+/** RenderableShapeBuilder builds various properties of a renderable.
+ * @author realitix */
+public class RenderableShapeBuilder extends BaseShapeBuilder {
+
+	private static class RenderablePool extends FlushablePool<Renderable> {
+		public RenderablePool () {
+			super();
+		}
+
+		@Override
+		protected Renderable newObject () {
+			return new Renderable();
+		}
+
+		@Override
+		public Renderable obtain () {
+			Renderable renderable = super.obtain();
+			renderable.environment = null;
+			renderable.material = null;
+			renderable.meshPart.set("", null, 0, 0, 0);
+			renderable.shader = null;
+			return renderable;
+		}
+	}
+
+	private static short[] indices = new short[1];
+	private static float[] vertices;
+	private final static RenderablePool renderablesPool = new RenderablePool();
+	private final static Array<Renderable> renderables = new Array<Renderable>();
+	private static final int FLOAT = 4;
```

Comment:
Is it even safe assuming a FLOAT is 4 bytes? Why not query OpenGL or the System, depending on the situation?